xsnippet_api/resources/snippets.py

```diff
@@ -16,44 +16,101 @@
 from ..application import endpoint
 
 
-_schema = {
-    'id': {'type': 'integer', 'readonly': True},
-    'title': {'type': 'string'},
-    'content': {'type': 'string', 'required': True},
-    'syntax': {'type': 'string'},
-    'tags': {'type': 'list', 'schema': {'type': 'string', 'regex': '[\w_-]+'}},
-    'is_public': {'type': 'boolean'},
-    'author_id': {'type': 'integer', 'readonly': True},
-    'created_at': {'type': 'datetime', 'readonly': True},
-    'updated_at': {'type': 'datetime', 'readonly': True},
-}
+class InvalidId(Exception):
+    pass
+
+
+def _get_id(resource):
+    v = cerberus.Validator({
+        'id': {'type': 'integer', 'min': 1, 'coerce': try_int},
+    })
+
+    if not v.validate(dict(resource.request.match_info)):
+        raise InvalidId('%s.' % cerberus_errors_to_str(v.errors))
+
+    return int(resource.request.match_info['id'])
+
+
+async def _write(resource, service_fn, *, status):
+    v = cerberus.Validator({
```

Comment:
I must have asked this before and it's probably not a big deal anyway, but can we create a Validator instance just once and reuse it rather than doing that on every call to _write()?